lib/mix/lib/mix/tasks/test.coverage.ex

```diff
@@ -0,0 +1,200 @@
+defmodule Mix.Tasks.Test.Coverage do
+  use Mix.Task
+
+  @moduledoc """
+  Build coverage report from exported coverage:
+
+      MIX_TEST_PARTITION=1 mix test --partitions 2
+      MIX_TEST_PARTITION=2 mix test --partitions 2
+      mix test.coverage
+
+  When using `--cover` with the default coverage tool,
+  it supports an `:export` option that exports the coverage
+  results into a directory. This is useful when there are
+  multiple test suites or when a single test suite is
+  partitioned across multiple runs when using the
+  `mix test --partitions N` option.
+
+  Once multiple test runs are exported, this task can be
+  used to generate an aggregated report.
+  """
+
+  @shortdoc "Build coverage report from exported coverage"
+  @recursive true
+  @preferred_cli_env :test
+  @default_threshold 90
+
+  @doc false
+  def run(_args) do
+    config = Mix.Project.config()
+    compile_path = Mix.Project.compile_path(config)
+    test_coverage = config[:test_coverage] || []
+    output = Keyword.get(test_coverage, :output, "cover")
+    pid = cover_compile(compile_path)
+
+    case Path.wildcard("#{output}/*.coverdata") do
+      [] ->
+        Mix.shell().error("Could not find any .coverdata file inside #{output}/")
+
+      entries ->
+        for entry <- entries do
+          Mix.shell().info("Importing cover results: #{entry}")
+          :ok = :cover.import(String.to_charlist(entry))
+        end
+
+        # Silence analyse import messages emitted by cover
+        {:ok, string_io} = StringIO.open("")
+        Process.group_leader(pid, string_io)
+
+        Mix.shell().info("")
+        generate_cover_results(test_coverage)
+    end
+  end
+
+  @doc false
+  def start(compile_path, opts) do
+    Mix.shell().info("Cover compiling modules ...")
+    cover_compile(compile_path)
+
+    if name = opts[:export] do
```

Comment:
For those willing to try this on v1.10, you can copy this whole module to your application and change only this line:

```suggestion
    if name = System.get_env("MIX_TEST_PARTITION") do
```

And then in your project `mix.exs`, you can do `test_coverage: [tool: Mix.Tasks.Test.Coverage]`.